tests/Unit/Helpers/DayOfEventTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Test\Unit\Helpers;
+
+use Engelsystem\Config\Config;
+use Engelsystem\Helpers\Carbon;
+use Engelsystem\Helpers\DayOfEvent;
+use Engelsystem\Test\Unit\ServiceProviderTest;
+
+class DayOfEventTest extends ServiceProviderTest
+{
+    private const FORMAT = 'Y-m-d H:i:s';
+    private Config $config;
+    private Carbon $now;
+
+    public function setUp(): void
+    {
+        $this->now = Carbon::createFromFormat(self::FORMAT, '2023-07-31 15:23:42');
+        Carbon::setTestNow($this->now);
+        $app = $this->createAndSetUpAppWithConfig([]);
+        $this->config = $app->get('config');
+    }
+
+    public function tearDown(): void
+    {
+        Carbon::setTestNow();
+    }
+
+    /**
+     * @return Array<string, array>
+     */
+    public function provideTestGetData(): array
+    {
+        return [
+            'no start date' => [null, true, null],
+
+            'day -2 (10:00, with day 0)' => [-2, true, Carbon::createFromFormat(self::FORMAT, '2023-07-28 10:00:00')],
+            'day -2 (23:59, with day 0)' => [-2, true,  Carbon::createFromFormat(self::FORMAT, '2023-07-28 23:59:59')],
+            'day -1 (23:59, with day 0)' => [-1, true,  Carbon::createFromFormat(self::FORMAT, '2023-07-29 23:59:59')],
+            'day 0 (with day 0)'        => [0, true, Carbon::createFromFormat(self::FORMAT, '2023-07-30 16:00:00')],
+            'day 1 (with day 0)'         => [1, true, Carbon::createFromFormat(self::FORMAT, '2023-07-31 10:00:00')],
+            'day 2 (with day 0)'         => [2, true, Carbon::createFromFormat(self::FORMAT, '2023-08-01 10:00:00')],
+
+            'day -2 (without day 0)'        => [
+                -2,
+                false,
+                Carbon::createFromFormat(self::FORMAT, '2023-07-29 10:00:00'),
```

Comment:
In general i would put the `Carbon::createFromFormat(self::FORMAT, [...]` inside the test method, that would make it easier to grasp

Reply:
Done